### PR TITLE
[IMP] l10n_in_pos: Set Place of Supply on Miscellaneous Journals

### PR DESCRIPTION
**Pull request by zepa-odoo**
In This PR:

- Set Place of Supply on 'general' journals with POS sessions if the country is India and Place of Supply is not set.
- Adjusted state ID computation for account moves with POS sessions.

Task Id: 3935317

### CODE REVIEW

File: addons/l10n_in_pos/models/account_move.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+from odoo import models
+
+
+class AccountMove(models.Model):
+    _inherit = 'account.move'
+
+    def _compute_l10n_in_state_id(self):
+        res = super()._compute_l10n_in_state_id()
+        for move in self:
```
Comment (jva-odoo) [inline addons/l10n_in_pos/models/account_move.py:10]:
```suggestion
        to_compute = self.filtered(lambda m: m.country_code == 'IN' and not m.l10n_in_state_id and m.journal_id.type == 'general')
        pos_session_moves = [pos['move_id'][0] for pos in self.env['pos.session'].sudo().search_read(
            domain= [('move_id', '=', to_compute.id)],
            fields= ['move_id'],
        )
        for move in to_compute:
```

File: addons/l10n_in_pos/models/account_move.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+            if not move.l10n_in_state_id and move.journal_id.type == 'general' and move.country_code == 'IN':
+                pos_session_count = self.env['pos.session'].sudo().search_count([('move_id', '=', move.id)])
+                if pos_session_count:
+                    move.l10n_in_state_id = move.company_id.state_id
```
Comment (jva-odoo) [inline addons/l10n_in_pos/models/account_move.py:14]:
```suggestion
            if move.id in pos_session_moves:
                move.l10n_in_state_id = move.company_id.state_id
```

**Review by jva-odoo — CHANGES_REQUESTED** [review]

File: addons/l10n_in_pos/models/account_move.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+from odoo import models
+
+
+class AccountMove(models.Model):
+    _inherit = 'account.move'
+
+    def _compute_l10n_in_state_id(self):
+        res = super()._compute_l10n_in_state_id()
+        to_compute = self.filtered(lambda m: m.country_code == 'IN' and not m.l10n_in_state_id and m.journal_id.type == 'general')
+        pos_session_moves = self.env['pos.session'].sudo().search_read(
```
Comment (jco-odoo) [inline addons/l10n_in_pos/models/account_move.py:11]: We can not move the l10n_in_pos_session_ids field to l10n_in_pos?  The ORM might handle it well.

Comment (jva-odoo) [reply addons/l10n_in_pos/models/account_move.py:11]: I don't think so because the new field needs an update of the module.
tell me if I am wrong.

Comment (jco-odoo) [reply addons/l10n_in_pos/models/account_move.py:11]: @jva-odoo It does not change the database model.  It is like a non-stored field.

Comment (zepa-odoo) [reply addons/l10n_in_pos/models/account_move.py]: @jco-odoo, I have moved the l10n_in_pos_session_ids field from l10n_in_reports_gstr_pos to l10n_in_pos. I have updated the code accordingly.

**Comment by davidmonnom** [comment]
@robodoo r+

**Pull request closed without merging**